geopandas/tests/test_geodataframe.py

```diff
@@ -1653,3 +1653,33 @@ def test_reduce_geometry_array():
     This warning is issued with pandas 2.2.2 (tested).
     """
     GeoDataFrame({"geometry": []}).all()
+
+
+class GDFChild(GeoDataFrame):
+    @property
+    def _constructor(self):
+        return GDFChild
+
+    def custom_method(self):
+        return "this is a custom output"
+
+
+def test_inheritance(dfs):
+    df, _ = dfs
+    df.loc[:, "col2"] = ["a"] * len(df)
+
+    dfc = GDFChild(df)
+
+    children = [
+        dfc,
+        dfc.iloc[[0]],
+        dfc.loc[dfc.col1 == 1],
+        dfc.rename_geometry("geometry2"),
+        dfc.dissolve(),
+        dfc[["col2", "geometry"]],
+        dfc.copy(),
+    ]
```

Comment:
ah that's not quite what I meant
```suggestion
    dfc2 = df2.rename_geometry("geometry2")
    children = [
        dfc,
        dfc.iloc[[0]],
        dfc.loc[dfc.col1 == 1],
        dfc.dissolve(),
        dfc[["col2", "geometry"]],
        dfc.copy(),
        dfc2,
        dfc2.iloc[[0]],
        dfc2.loc[dfc.col1 == 1],
        dfc2.rename_geometry("geometry2"),
        dfc2.dissolve(),
        dfc2[["col2", "geometry"]],
        dfc2.copy(),
    ]
```